src/Engine/Manager/ParticleManager.cpp

```diff
@@ -80,80 +57,47 @@ int ParticleManager::GetTextureAtlasRows() const {
-    
+Component::ParticleSystemComponent* ParticleManager::CreateParticleSystem(const Json::Value& node) {
+
+    Component::ParticleSystemComponent* particleSystem = InitParticleSystem(particleSystems.Create());
     // Load values from Json node.
-    particleEmitter->particleType.textureIndex = node.get("textureIndex", 0).asInt();
-    particleEmitter->particleType.minVelocity = Json::LoadVec3(node["minVelocity"]);
-    particleEmitter->particleType.maxVelocity = Json::LoadVec3(node["maxVelocity"]);
-    particleEmitter->particleType.averageLifetime = node.get("averageLifetime", 0.f).asFloat();
-    particleEmitter->particleType.lifetimeVariance = node.get("lifetimeVariance", 0.f).asFloat();
-    particleEmitter->particleType.averageSize = Json::LoadVec2(node["averageSize"]);
-    particleEmitter->particleType.sizeVariance = Json::LoadVec2(node["sizeVariance"]);
-    particleEmitter->particleType.uniformScaling = node.get("uniformScaling", false).asBool();
-    particleEmitter->particleType.startAlpha = node.get("startAlpha", 0.f).asFloat();
-    particleEmitter->particleType.midAlpha = node.get("midAlpha", 1.f).asFloat();
-    particleEmitter->particleType.endAlpha = node.get("endAlpha", 0.f).asFloat();
-    particleEmitter->particleType.color = Json::LoadVec3(node["color"]);
-    particleEmitter->size = Json::LoadVec3(node["size"]);
-    particleEmitter->averageEmitTime = node.get("averageEmitTime", 0.03).asFloat();
-    particleEmitter->emitTimeVariance = node.get("emitTimeVariance", 0.03).asFloat();
-    particleEmitter->emitterType = static_cast<Component::ParticleEmitter::EmitterType>(node.get("emitterType", 0).asInt());
-    
-    return particleEmitter;
+    particleSystem->particleType.textureIndex = node.get("textureIndex", 0).asInt();
+    particleSystem->particleType.nr_new_particles = node.get("emitAmount", 8).asInt();
+    particleSystem->particleType.rate = node.get("rate", 0.3).asFloat();
+    particleSystem->particleType.lifetime = node.get("lifetime", 10.0).asFloat();
+    particleSystem->particleType.scale = node.get("scale", 10.0).asFloat();
+    particleSystem->particleType.velocity = Json::LoadVec3(node["velocity"]);
+    particleSystem->particleType.alpha_control = node.get("alphaControl", 10.0).asFloat();
+    particleSystem->particleType.mass = node.get("mass", 1.0).asFloat();
+    particleSystem->particleType.spread = node.get("spread", 1.0).asFloat();
+    particleSystem->particleType.randomVec = Json::LoadVec3(node["randomVelocity"]);
+    particleSystem->particleType.velocityMultiplier = node.get("speed", 10.0).asFloat();
+    particleSystem->particleType.nr_particles = node.get("NrOfParticles", 1024).asInt();
+
+    return particleSystem;
 }
 
-const std::vector<Component::ParticleEmitter*>& ParticleManager::GetParticleEmitters() const {
-    return particleEmitters.GetAll();
+void ParticleManager::RemoveParticleRenderer(Component::ParticleSystemComponent * component) {
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') The function 'RemoveParticleRenderer' is never used. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://localhost:9000/coding_rules#rule_key=cppcheck%3AunusedFunction)